lobsterpy/featurize/core.py

```diff
@@ -0,0 +1,829 @@
+                    e_range=self.e_range,
+                )
+                / self._get_n_moment(
+                    n=2,
+                    coxx=coxx_dict[self.feature_type],
+                    energies=energies,
+                    e_range=self.e_range,
+                )
+                ** 2
+            )
+        except KeyError:
+            raise ValueError(
+                "Please recheck fp_type requested argument.Possible options are bonding/antibonding/overall"
+            )
+
+        return coxx_center, coxx_width, coxx_skew, coxx_kurt
+
+    def _get_coxx_center(
+        self,
+        coxx: npt.NDArray[np.floating],
+        energies: npt.NDArray[np.floating],
+        e_range: List[float],
+    ) -> float:
+        """
+        Get the band width, defined as the first moment of the orbital resolved COHP
+        Args:
+            cohp: Orbital COHP list
+            energies: energies corresponding orbital orbital interactions COHP
+        Returns:
+            Orbital-Orbital interaction band center in eV
+        """
+        coxx_center = self._get_n_moment(
+            n=1, coxx=coxx, energies=energies, e_range=e_range, center=False
+        )
+
+        return coxx_center
+
+    def _get_n_moment(
+        self,
+        n: float,
+        coxx: npt.NDArray[np.floating],
+        energies: npt.NDArray[np.floating],
+        e_range: list[float] | None,
+        center: bool = True,
+    ) -> float:
+        """
+
+        Get the nth moment of COXX
+
+        Args:
+            n: The order for the moment
+            cohp: COXX list
+            energies: energies
+            center: Take moments with respect to the COXX center
+
+        Returns:
+            COXX nth moment in eV
+        """
+        if e_range:
+            coxx = coxx[(energies >= self.e_range[0]) & (energies <= self.e_range[-1])]
+            energies = energies[
+                (energies >= self.e_range[0]) & (energies <= self.e_range[-1])
+            ]
+
+        if center:
+            coxx_center = self._get_coxx_center(
+                coxx=coxx, energies=energies, e_range=self.e_range
+            )
+            p = energies - coxx_center
+        else:
+            p = energies
+
+        nth_moment = np.trapz(p**n * coxx, x=energies) / np.trapz(coxx, x=energies)  # type: ignore
+
+        return nth_moment
+
+    def get_summarized_coxx_df(
+        self, ids: str | None = None, label_list: List[str] | None = None
+    ) -> pd.DataFrame:
+        """
+        This function returns a pandas dataframe with weighted ICOXX, effective interaction number
+        and moment features (center, width, skewness and kurtosis) of COXX in selected energy range
+
+        Returns:
+            Returns a pandas dataframe with cohp/cobi/coop related features as per input file
+
+        """
+        if ids:
+            df = pd.DataFrame(index=[ids])
+        else:
+            ids = os.path.basename(os.path.dirname(self.path_to_coxxcar))
```

Comment:
In many parts of the code, we now use "Path". Could you update this?

Reply:
Yes